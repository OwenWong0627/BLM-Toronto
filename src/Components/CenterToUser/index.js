import React from 'react';
import PropTypes from 'prop-types';

/**
 * This component will be a small icon placed on the Google Map that will take in a panTo function so when it is clicked,
 * it will "pan to" to the center prop which is also passed into this component
 */
function CenterToUser({ panTo, center }) {
   return (
      <button
         data-testid="centerToUser-button"
         className="centerToUser"
         onClick={() => {
            panTo({
               lat: center.lat,
               lng: center.lng,
            });
         }}
      >
         <img src="/HomeMarker.svg" alt="Home Marker" />
      </button>
   );
}

CenterToUser.propTypes = {
   panTo: PropTypes.func.isRequired,
   center: PropTypes.object.isRequired
}

export default CenterToUser;