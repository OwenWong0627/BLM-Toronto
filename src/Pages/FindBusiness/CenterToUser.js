import React from "react";

//Center to User's Location Button
function CenterToUser({ panTo, center }) {
   return (
      <button
         className="centerToUser"
         onClick={() => {
            panTo({
               lat: center.lat,
               lng: center.lng,
            });
         }}
      >
         <img src="/compass.svg" alt="compass" />
      </button>
   );
}

export default CenterToUser;