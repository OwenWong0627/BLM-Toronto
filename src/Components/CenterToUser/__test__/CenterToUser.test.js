import React from 'react';
import ReactDOM from 'react-dom';
import CenterToUser from '..';

import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

const mockCenter = { lat: 43.1, lng: -79.6 };

afterEach(cleanup);
it('Renders without crashing', () => {
   const div = document.createElement("div");
   ReactDOM.render(<CenterToUser />, div);
});

test('if the onClick in the component is responsive', () => {
   const mockFunction = jest.fn();
   const { getByTestId } = render(<CenterToUser panTo={mockFunction} center={mockCenter} />);
   fireEvent.click(getByTestId("centerToUser-button"));
   expect(mockFunction.mock.calls[0][0]).toEqual(mockCenter);
});

it('matches snapshot', () => {
   const tree = renderer.create(<CenterToUser />).toJSON();
   expect(tree).toMatchSnapshot();
});