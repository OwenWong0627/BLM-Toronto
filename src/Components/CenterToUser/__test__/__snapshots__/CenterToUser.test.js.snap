// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="centerToUser"
  data-testid="centerToUser-button"
  onClick={[Function]}
>
  <img
    alt="Home Marker"
    src="/HomeMarker.svg"
  />
</button>
`;
